packages/ListBox/src/components/Options/helpers/options.js

```diff
@@ -267,10 +303,14 @@ export function handleEnterOrSpace({ event, providedProps, state, dispatch, onCh
     }
 
     // for single select the option is set when the user interact with up and down arrows
-    // no need to notify which option is selected just close the popover
-    dispatch({ type: useListBox.types.closePopover });
-    if (state.refTrigger.current) {
-      state.refTrigger.current.focus();
+    // no need to notify which option is selected just close the popover unless the option is disabled
+    const id = state.optionsIndex[document.activeElement.getAttribute("id")];
+    if (!state.options[id].isDisabled) {
+      dispatch({ type: useListBox.types.closePopover });
+
+      if (state.refTrigger.current) {
+        state.refTrigger.current.focus();
+      }
```

Comment:
Can we do `state.refTrigger.current?.focus();` 🤔

Reply:
🤷‍♂️ just different flavour of 🍨